.vsts-pipelines/templates/steps/script.yml

```diff
@@ -0,0 +1,70 @@
+# File: templates/steps/script.yml
+
+parameters:
+  name: 'DO NOT LEAVE DEFAULT'
+  agentOs: 'Windows'
+  beforeBuild: []
+  buildScript: ''
+  buildArgs: ''
+  afterBuild: []
+  hasTests: false
+  autoArtifact: true
+  envVars:
+    BuildNumber: $(Build.BuildNumber)
+    CI: true
+    DOTNET_CLI_TELEMETRY_OPTOUT: 1
+    DOTNET_HOME: '$(Build.SourcesDirectory)/.dotnet'
+    DOTNET_SKIP_FIRST_TIME_EXPERIENCE: true
+
+steps:
+  - ${{ parameters.beforeBuild }}
+  - ${{ if endsWith(parameters.buildScript, '.ps1') }}:
+    - task: Powershell@1
+      displayName: ${{ parameters.buildScript }}
+      env:
+        ${{ parameters.envVars }}
+      inputs:
+        name:
+        scriptType: filePath
+        scriptName: ${{ parameters.buildScript }}
+        arguments: ${{ parameters.buildArgs }}
+  - ${{ if endsWith(parameters.buildScript, '.sh') }}:
+    - task: Bash@3
+      displayName: ${{ parameters.buildScript }}
+      env:
+        ${{ parameters.envVars }}
+      inputs:
+        scriptType: filePath
+        filePath: ${{ parameters.buildScript }}
+        arguments: ${{ parameters.buildArgs }}
+  - ${{ if endsWith(parameters.buildScript, '.cmd') }}:
+    - task: BatchScript@1
+      displayName: ${{ parameters.buildScript }}
+      env:
+        ${{ parameters.envVars }}
+      inputs:
+        filename: ${{ parameters.buildScript }}
+        arguments: ${{ parameters.buildArgs }}
+  - ${{ if eq(parameters.hasTests, true) }}:
+    - task: PublishTestResults@2
+      displayName: Publish test results
+      condition: always()
+      inputs:
+        testRunner: vstest
+        testResultsFiles: 'modules/*/artifacts/logs/*.trx'
+  - ${{ parameters.afterBuild }}
+  - ${{ if eq(parameters.autoArtifact, true) }}:
+    - task: PublishBuildArtifacts@1
+      displayName: Upload artifacts
+      inputs:
+        pathtoPublish: ./artifacts/
+        artifactName: ${{ parameters.name }}
+        artifactType: Container
+
+
+    # ${{ if and(eq(parameters.buildScript, ''), eq(parameters.agentOs, 'Windows')) }}:
```

Comment:
Did you try something like this? https://github.com/aspnet/BuildTools/blob/3260c92114dd466700f24301e8e52d007767c54a/.vsts-pipelines/templates/phases/default-build.yml#L36-L41